### PR TITLE
Made Art Feature Responsive and Added Instructions Modal

### PR DESCRIPTION
**Pull request by rushabh-t**
#####

- Made the page responsive
- Added Instructions Modal

#####

Before :

Capture

After :

The page with the instruction button:
Capture1

Instruction Modal:
Capture2

Tab view:
Capture3

Mobile view:
Capture4

### CODE REVIEW

File: js/artFeature.js [hunk 1]
```diff
@@ -17,3 +17,15 @@ const creatingArtFromHtml = () => {
 document
   .querySelector('#html-code')
   .addEventListener('keyup', creatingArtFromHtml);
+
+const instructionButton = document.getElementById('instruction-button');
+const closeButton = document.getElementsByClassName('close')[0];
+const instructionModal = document.getElementById('instruction-modal');
```
Comment (rohan09-raj) [inline js/artFeature.js:23]: Can we please move these assignments to top of the file ?

File: css/art-feature.css [hunk 2]
```diff
@@ -67,8 +67,91 @@ input {
...
+  top: 0;
+  width: 100%;
+  height: 100%;
+  background-color: rgb(0, 0, 0);
+  background-color: rgba(0, 0, 0, 0.4);
+}
+
+.instruction-content {
+  background-color: #fefefe;
+  margin: 7%;
```
Comment (rohan09-raj) [inline css/art-feature.css:137]: can we please use `px` here ??

File: art-feature.html [hunk 3]
```diff
@@ -6,12 +6,31 @@
     <link rel="stylesheet" href="./css/art-feature.css" />
   </head>
   <body>
-    <h1 class="welcome-title">Your Art</h1>
+    <div id="title-container">
```
Comment (rohan09-raj) [inline art-feature.html:9]: Can we please make this as a class ?? and use `query selector` in the JS file

**Comment by rushabh-t** [comment]
I have made all the changes as discussed and also the ones mentioned here. One thing I wanted to add is, I have made the instruction modal scrollable, as it occupied more height and had to make it scrollable.

File: js/artFeature.js [hunk 4]
```diff
@@ -17,3 +23,23 @@ const creatingArtFromHtml = () => {
...
+instructionButton.onclick = () => {
+  instructionModal.style.display = overlay.style.display = 'block';
+  document.querySelector('body').style.overflow = 'hidden';
+  if (mobileScreen.matches) {
+    instructionModal.style.height = '350px';
+  }
+};
+
+closeButton.onclick = () => {
+  instructionModal.style.display = overlay.style.display = 'none';
```
Comment (DeRaowl) [inline js/artFeature.js:36]: If you are handling modal with display: none you cannot do transitions. How are you handling the transition of open and close of modal?

**Review by DeRaowl — CHANGES_REQUESTED** [review]

**Review by rohan09-raj — COMMENTED** [review]
@rushabh-t Rebasing has not been done properly. There should not be previous commits added here.
Also please write specific commit messages according to the changes made.

File: css/art-feature.css [hunk 5]
```diff
@@ -67,8 +73,120 @@ input {
...
+#instruction-button:hover {
+  color: white;
+  box-shadow: 0 12px 16px 0 rgba(0, 0, 0, 0.24),
+    0 17px 50px 0 rgba(0, 0, 0, 0.19);
+}
+
+.instruction-modal {
+  display: none;
+  position: absolute;
+  padding-left: 1rem;
```
Comment (rohan09-raj) [inline css/art-feature.css:112]: Can you please use `px` here for consistency ?

**Review by rohan09-raj — CHANGES_REQUESTED** [review]

File: css/art-feature.css [hunk 6]
```diff
@@ -67,8 +73,120 @@ input {
...
+  font-size: 1rem;
+}
+
+.title-container {
+  display: flex;
+  flex-direction: column;
+}
+
+#instruction-button {
+  position: relative;
```
[the remark this answers is not in the document]
Comment (rushabh-t) [reply css/art-feature.css:88]: Removed it, thanks for pointing it out

File: art-feature.html [hunk 7]
```diff
@@ -1,17 +1,45 @@
...
+    <div class="overlay">
+      <div class="instruction-modal">
+        <button class="close">&times;</button>
+        <h1>Instructions</h1>
+        <p>
+          Hey User! Welcome to CSS Art Feature<br /><br />
+          These are few intructions which you need to follow strictly for
+          presenting your creativity else you won't be getting the desired
+          results:<br /><br />
+          ⦁ Use only CSS format style code.<br />
```
[the remark this answers is not in the document]
Comment (rushabh-t) [reply art-feature.html:27]: Sure, will make the change.

File: css/art-feature.css [hunk 8]
```diff
@@ -67,8 +73,120 @@ input {
...
+  width: 110px;
+  margin-top: 20px;
   border-radius: 5px;
   padding: 15px;
   font-size: 1rem;
+  transition-duration: 0.4s;
+}
+
+#instruction-button:hover {
+  color: white;
```
[the remark this answers is not in the document]
Comment (rushabh-t) [reply css/art-feature.css:104]: removed

File: js/artFeature.js (hunk 4 above)
Comment (rushabh-t) [reply js/artFeature.js:36]: I haven't used any transition for open/close of modal. Should I add it?

File: css/art-feature.css (hunk 5 above)
Comment (rushabh-t) [reply css/art-feature.css:112]: Made this change

File: js/artFeature.js [hunk 9]
```diff
@@ -17,3 +23,23 @@ const creatingArtFromHtml = () => {
 document
   .querySelector('#html-code')
   .addEventListener('keyup', creatingArtFromHtml);
+
+instructionButton.onclick = () => {
+  instructionModal.style.display = overlay.style.display = 'block';
+  document.querySelector('body').style.overflow = 'hidden';
+  if (mobileScreen.matches) {
+    instructionModal.style.height = '350px';
```
[the remark this answers is not in the document]
Comment (rushabh-t) [reply js/artFeature.js:31]: Done the change.

File: css/art-feature.css [hunk 10]
```diff
@@ -2,6 +2,13 @@ body {
   font-family: 'Roboto', sans-serif;
 }
 
+img {
```
[the remark this answers is not in the document]
Comment (rushabh-t) [reply css/art-feature.css:5]: Done, used a class selector.

File: css/art-feature.css [hunk 11]
```diff
@@ -67,8 +73,120 @@ input {
...
+  cursor: pointer;
+  font-weight: bold;
+  border: none;
+  height: 50%;
+  width: 110px;
+  margin-top: 20px;
   border-radius: 5px;
   padding: 15px;
   font-size: 1rem;
+  transition-duration: 0.4s;
```
[the remark this answers is not in the document]
Comment (DeRaowl) [reply css/art-feature.css:100]: transition-duration is added but you should add on property name too.

[the remark this answers is not in the document]
Comment (rushabh-t) [reply css/art-feature.css:99]: Added

**Review by rohan09-raj — APPROVED** [review]
Awesome 🔥, good work 🎉  @rushabh-t
Approving and merging your PR

**Pull request merged**